### ISSUE TITLE
Attempting to use `ruff check` with includes, but without specifying a directory, picks up nested projects

### ISSUE THREAD

**Issue by eli-schwartz** [astral-sh/ruff#9019]
... such as git worktree clones.

Given a .ruff.toml with the contents:

```
include = ["mesonbuild/**/*.py"]
```

```
$ git worktree add -d bisect/
$ ruff check
```

Every possible error is listed twice, once for the mesonbuild/ directory and once for the bisect/mesonbuild/ directory.

This only seems to happen if the worktree has a .ruff.toml as well (which it will if it is tracked in git).

Using `ruff check mesonbuild/` the traditional way continues to work fine.

**Comment by MichaReiser** [comment]
Can you try using `include = ["/mesonbuild/**/*.py"] to make it explicit, that the pattern should only match `mesonbuild` directories at the root?

**Comment by charliermarsh** [comment]
This _sort of_ makes sense given that `include` is only applied at the file level... So we visit each file within `bisect/mesonbuild`, and each of those files will resolve to the configuration at `bisect/mesonbuild/.ruff.toml`, which says that they should be included. (I'm not saying it's the right behavior, but I can at least understand why it's happening.)

**Comment by eli-schwartz** [comment]
That's actually the first thing I tried as a wild guess.

```
warning: No Python files found under the given path(s)
```

Afterwards I realized that it was conditional on the subdirectory also having a .ruff.toml, which made me rethink whether that was related at all.

**Comment by bswck** [comment]
Here's the workaround I use.

```bash
linked-worktrees() {
    git worktree list --porcelain | rg "worktree (.+)" -r '$1' | tail -n +2
}

ruff() {
    local RUFF="$(which ruff)"
    # shellcheck disable=SC2046,SC2048,SC2086
    case "$1" in
        check) "$RUFF" check $(linked-worktrees | xargs -l echo --extend-exclude) ${*:3};;
        *) "$RUFF" ${*:1};;
    esac
}
```

to be put in a file like `~/.bashrc`.

**Comment by AAraKKe** [comment]
I think it would make a lot of sense when ruff navigates the files to format/lint to allow excluding worktrees. Those should be very easily detectable since we just need to check the `.git` folder or even ask git about them. People use worktrees often to navigate from one state of the project to another one and it seems rather straightforward to have ruff detecting them.

I cannot think of an usecase where you would like to format every python file in all your worktrees but, in case this is something you might want to do, this could be configurable with the default being to ignore them as they are, in essence, separate projects.
